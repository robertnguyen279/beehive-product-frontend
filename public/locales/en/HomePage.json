{
  "welcome": "Welcome to Beehive",
  "login-button": "Login",
  "join-club": "Join the club",
  "join-club-description": "Join us now to explore new social network experience",
  "login-left-community": "Community",
  "login-left-community-description": "Share your experience with big community",
  "login-left-onlineshop": "Online Shop",
  "login-left-onlineshop-description": "Explore our big online shopping system",
  "login-left-jobsearch": "Job search",
  "login-left-jobsearch-description": "Search for jobs near your location"
}
