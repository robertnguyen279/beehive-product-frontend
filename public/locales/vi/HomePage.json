{
  "welcome": "Chào mừng với Beehive",
  "login-button": "Đăng nhập",
  "join-club": "Tham gia với chúng tôi",
  "join-club-description": "Tham gia ngay để trải nghiệm mạng xã hội hoàn toàn mới",
  "login-left-community": "Cộng đồng",
  "login-left-community-description": "Chia sẻ trải nghiệm của bạn với cộng đồng lớn mạnh",
  "login-left-onlineshop": "Online Shop",
  "login-left-onlineshop-description": "Mua sắm online thỏa thích",
  "login-left-jobsearch": "Việc làm",
  "login-left-jobsearch-description": "Tìm kiếm việc làm quanh bạn"
}
