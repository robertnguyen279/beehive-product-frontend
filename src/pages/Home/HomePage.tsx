import React from 'react';
// import { useTranslation } from 'react-i18next';
import { useMediaQuery } from 'react-responsive';
import HeaderHome from './Header';
import LoginContainer from './LoginContainer';

function HomePage(): React.ReactElement {
  // const { t } = useTranslation('HomePage');
  const isLaptop = useMediaQuery({ query: '(min-width: 768px)' });

  return (
    <React.Fragment>
      <div className="home__container min-h-screen min-w-full relative">
        <HeaderHome isLaptop={isLaptop} />
        <LoginContainer isLaptop={isLaptop} />
        <div className="home__layer--dark absolute w-full h-full inset-0"></div>
      </div>
    </React.Fragment>
  );
}

export default HomePage;
