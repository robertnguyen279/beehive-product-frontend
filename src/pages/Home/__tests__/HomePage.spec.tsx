import React from 'react';
import { Provider } from 'react-redux';
import HomePage from '../HomePage';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
// const clickFn = jest.fn();
const mockStore = configureStore([]);
const store = mockStore({});

describe('HomePage', () => {
  const component = shallow(
    <Provider store={store}>
      <HomePage />
    </Provider>,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
