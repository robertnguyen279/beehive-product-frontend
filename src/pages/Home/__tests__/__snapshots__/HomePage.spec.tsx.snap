// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly 1`] = `
<div
  className="home__container h-screen min-w-full relative"
  style={
    Object {
      "background": "url('test-file-stub')",
    }
  }
>
  <HeaderHome />
  <LoginContainer />
  <HomeSlider />
  <Footer />
  <div
    className="home__layer absolute w-full h-20 md:h-full inset-0"
  />
</div>
`;
